### PR TITLE
feat: update thirdparty in macOS

### PR DESCRIPTION
**Pull request by jingchen2222**
* **Please check if the PR fulfills these requirements**

    - The commit message follows our guidelines
    - Tests for the changes have been added (for bug fixes / features)
    - Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
update

* **What is the current behavior?** (You can also link to an open issue here)
#59 

* **What is the new behavior (if this is a feature change)?**
  - change std c++ 17
  - add zetasql libraries into project
  - add zetasql parser test
  - move `engine_test_base` and `test_base` to `src/testing`
  - build test with hybridse_core_shared lib in Linux to reduce test bins size

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -149,6 +152,56 @@ include_directories(${PROJECT_SOURCE_DIR}/include)
 link_directories(${PROJECT_SOURCE_DIR}/thirdparty/lib)
 link_directories(${PROJECT_SOURCE_DIR}/thirdparty/lib64)
 
+
+set(absl_ROOT "${PROJECT_SOURCE_DIR}/thirdparty")
```
Comment (aceforeverd) [inline CMakeLists.txt:156]: try use `CMAKE_PREFIX_PATH`

**Comment by jingchen2222** [comment]
Build test case with hybridse_core_shared. This can **Largely** reduce test bins size

File: .github/workflows/hybridse-ci.yml [hunk 2]
```diff
@@ -281,29 +289,29 @@ jobs:
...
-          ./tools/compile_and_coverage.sh
-
-      - name: Upload Coverage Report
-        uses: codecov/codecov-action@v1
-        with:
-          files: build/coverage.info
-          name: coverage
-          fail_ci_if_error: true
-          verbose: true
+#  coverage:
```
Comment (aceforeverd) [inline .github/workflows/hybridse-ci.yml:292]: can coverage enable now ?

**Comment by aceforeverd** [comment]
> Build test case with hybridse_core_shared. This can **Largely** reduce test bins size.
> This might Resolve #94 only if we can handle the crush during`make test`

seems there are many debug section
use `strip --strip-unneeded` may help as well

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -77,15 +86,20 @@ if (TESTING_ENABLE AND CORE_TESTING_ENABLE)
...
+			${yaml_libs} ${BRPC_LIBS})
+		endif()
+        add_test(${TEST_TARGET_NAME}
+				${CMAKE_CURRENT_BINARY_DIR}/${TEST_TARGET_DIR}/${TEST_TARGET_NAME}
+				--gtest_output=xml:${CMAKE_BINARY_DIR}/${TEST_TARGET_NAME}.xml)
 
-            target_link_options(${TEST_TARGET_NAME} PRIVATE -rdynamic)
-	    target_link_libraries(${TEST_TARGET_NAME} gtest hybridse_core hybridse_flags
-                ${yaml_libs} ${BRPC_LIBS})
+		target_link_options(${TEST_TARGET_NAME} PRIVATE -rdynamic)
```
Comment (aceforeverd) [inline src/CMakeLists.txt:102]: what does `rdynamic` mean ?

File: examples/toydb/src/CMakeLists.txt [hunk 4]
```diff
@@ -60,7 +60,9 @@ if (TESTING_ENABLE AND EXAMPLES_TESTING_ENABLE)
         get_filename_component(TEST_TARGET_DIR ${RELATIVE_TEST_PATH} DIRECTORY)
         get_filename_component(TEST_TARGET_NAME ${RELATIVE_TEST_PATH} NAME_WE)
         set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${TEST_TARGET_DIR})
-        add_executable(${TEST_TARGET_NAME} ${TEST_SCRIPT})
+        add_executable(${TEST_TARGET_NAME} ${TEST_SCRIPT}
```
Comment (aceforeverd) [inline examples/toydb/src/CMakeLists.txt:63]: summary?:
+ link shared for test in src
+ link static for test in example

**Review by jingchen2222 — COMMENTED** [review]
> > Build test case with hybridse_core_shared. This can **Largely** reduce test bins size.
> > This might Resolve #94 only if we can handle the crush during`make test`
> 
> seems there are many debug section
> use `strip --strip-unneeded` may help as well

Yep maybe we can try it later.

File: examples/toydb/src/CMakeLists.txt (hunk 4 above)
Comment (jingchen2222) [reply examples/toydb/src/CMakeLists.txt:63]: Yep. I haven't try shared in the `example` module. Maybe we can try it in the next pr.

File: src/CMakeLists.txt (hunk 3 above)
Comment (jingchen2222) [reply src/CMakeLists.txt:102]: In fact, I don't know.

**Review by aceforeverd — APPROVED** [review]

**Pull request merged**